File: week2/day1/SLL.js

class SinglyLinkedList {
    constructor() { // def __init__()
        this.head = null
        this.length = 0
    }

    // Methods

    // Add to Back
    addToBack(val) {
        const newNode = new ListNode(val);

        if (this.head === null) {
            this.head = newNode;
            return this;
        }

        let runner = this.head;

        while (runner.next) {
            runner = runner.next;
        }

        runner.next = newNode;
        return this
    }
    // Add to front
    addToFront(val) {
        const newNode = new ListNode(val);
        const tempHead = this.head;
        this.head = newNode;
        this.head.next = tempHead;
    }

    // Remove from Front
    // Remove from Back

}

class ListNode {
    constructor(val) {
        this.value = val
        this.next = null
    } 
}

const SLL = new SinglyLinkedList();
SLL.addToBack(1);
SLL.addToBack(2);
SLL.addToBack(3);
SLL.addToBack(4);
SLL.addToBack(5);
SLL.addToBack(6);
console.log(SLL.head);

// Add and Remove Your nodes

File: week2/day1/SLL.py
class SinglyLinkedList:
    def __init__(self):
        self.head = None
        self.length = 0

    # Methods

    # Add to Back
    def addToBack(self,val):
        newNode = ListNode(val)

        if self.head == None:
            self.head = newNode
            return self;

        runner = self.head;

        while runner.next:
            runner = runner.next;

        runner.next = newNode;
        return self
        
    # Add to front
    def addToFront(self,val):
        newNode = ListNode(val)
        tempHead = self.head
        self.head = newNode
        self.head.next = tempHead

    # Remove from Front
    # Remove from Back

class ListNode:
    def __init__(self,val):
        self.value = val
        self.next = None


SLL = SinglyLinkedList();
# Add and Remove Your nodes